packages/rest/src/controllers/proofs/ProofController.ts

```diff
@@ -0,0 +1,236 @@
+import { Agent, PresentationPreview, RecordNotFoundError } from '@aries-framework/core'
+import { JsonEncoder } from '@aries-framework/core/build/utils/JsonEncoder'
+import {
+  Body,
+  Delete,
+  Get,
+  InternalServerError,
+  JsonController,
+  NotFoundError,
+  OnUndefined,
+  Param,
+  Post,
+} from 'routing-controllers'
+import { Inject, Service } from 'typedi'
+
+import { AcceptProofProposalRequest } from '../../schemas/AcceptProofProposalRequest'
+import { ProofRequestRequest } from '../../schemas/ProofPresentationRequest'
+import { ProofProposalRequest } from '../../schemas/ProofProposalRequest'
+import { PresentationProofRequestRequest } from '../../schemas/SendProofPresentationRequest'
+import { ProofUtils } from '../../utils/ProofUtils'
+
+@JsonController('/proofs')
+@Service()
+export class ProofController {
+  @Inject()
+  private agent: Agent
+  private proofUtils: ProofUtils
+
+  public constructor(agent: Agent) {
+    this.agent = agent
+    this.proofUtils = new ProofUtils(agent)
+  }
+
+  /**
+   * Retrieve proof record by proofRecordId
+   */
+  @Get('/:proofRecordId')
+  public async getProofById(@Param('proofRecordId') proofRecordId: string) {
+    try {
+      const proof = await this.agent.proofs.getById(proofRecordId)
+
+      return proof.toJSON()
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`proof with proofRecordId "${proofRecordId}" not found.`)
+      }
+      throw new InternalServerError(`something went wrong: ${error}`)
+    }
+  }
+
+  /**
+   * Retrieve proof record by threadId
+   */
+  @Get('/thread/:threadId')
+  public async getProofByThreadId(@Param('threadId') threadId: string) {
+    try {
+      const proof = await this.proofUtils.getProofByThreadId(threadId)
```

Comment:
We could have two records for the same thread id (meaning threadId is not unique). So I'd rather make the threadId an optional search param to getAllProofs